contracts/voting_escrow/src/contract.rs

```diff
@@ -60,13 +64,48 @@ pub fn instantiate(
 ) -> Result<Response, ContractError> {
     set_contract_version(deps.storage, CONTRACT_NAME, CONTRACT_VERSION)?;
 
+    // Accept values within [0,1] limit. If max_exit_penalty is empty then 100% penalty is considered.
+    let max_exit_penalty = msg
+        .max_exit_penalty
+        .map(|penalty| {
+            if penalty > Decimal::one() {
+                Err(StdError::generic_err("Max exit penalty should be <= 1"))
+            } else {
+                Ok(penalty)
+            }
+        })
+        .transpose()?
+        .unwrap_or_else(Decimal::one);
+    let slashed_fund_receiver = msg
+        .slashed_fund_receiver
+        .map(|addr| addr_validate_to_lower(deps.api, &addr))
+        .transpose()?;
+
     let config = Config {
         owner: addr_validate_to_lower(deps.api, &msg.owner)?,
         guardian_addr: addr_validate_to_lower(deps.api, &msg.guardian_addr)?,
         deposit_token_addr: addr_validate_to_lower(deps.api, &msg.deposit_token_addr)?,
+        max_exit_penalty,
+        slashed_fund_receiver,
     };
     CONFIG.save(deps.storage, &config)?;
 
+    // Initialize early withdraw parameters
+    let xastro_minter_resp: MinterResponse = deps
+        .querier
+        .query_wasm_smart(&config.deposit_token_addr, &Cw20QueryMsg::Minter {})?;
+    let staking_config: astroport::staking::ConfigResponse = deps.querier.query_wasm_smart(
+        &xastro_minter_resp.minter,
+        &astroport::staking::QueryMsg::Config {},
+    )?;
+    WITHDRAWAL_PARAMS.save(
+        deps.storage,
+        &WithdrawalParams {
+            astro_addr: staking_config.deposit_token_addr,
+            staking_addr: Addr::unchecked(xastro_minter_resp.minter),
+        },
+    )?;
+
```

Comment:
Maybe just pass it as parameters and store it in the Config?

Reply:
Passing this as parameters is error prone while doing it by contract logic is robust. Anyway this action will happen only once (during contract initialization or migration) so we won't spend gas later.

Reply:
Ok, but we always additionally read WITHDRAWAL_PARAMS paying some gas, let's move it to the Config?

Reply:
Ok, done